src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1634,24 +1634,25 @@ private bool IsPartialType()
 
         private bool IsPartialMember()
         {
-            // note(cyrusn): this could have been written like so:
-            //
-            //  return
-            //    this.CurrentToken.ContextualKind == SyntaxKind.PartialKeyword &&
-            //    this.PeekToken(1).Kind == SyntaxKind.VoidKeyword;
-            //
-            // However, we want to be lenient and allow the user to write 
-            // 'partial' in most modifier lists.  We will then provide them with
-            // a more specific message later in binding that they are doing 
-            // something wrong.
-            //
-            // Some might argue that the simple check would suffice.
-            // However, we'd like to maintain behavior with 
-            // previously shipped versions, and so we're keeping this code.
+            Debug.Assert(this.CurrentToken.ContextualKind == SyntaxKind.PartialKeyword);
```

Comment:
Based on internal discussion thread I was expecting to see some check of `IsFeatureEnabled(MessageID.IDS_FeaturePartialConstructorsAndEvents)` in here, a la the check for `IDS_FeatureRecords` above.